writer.go

```diff
@@ -65,12 +70,60 @@ func NewWriterLevel(w io.Writer, compressionLevel int) *Writer {
 //
 // Call Release when the Writer is no longer needed.
 func NewWriterDict(w io.Writer, cd *CDict) *Writer {
-	return newWriterDictLevel(w, cd, 0)
+	params := &WriterParams{
+		Dict: cd,
+	}
+
+	return newWriterParams(w, params)
+}
+
+const (
+	// WindowLogMin is the minimum value of the windowLog parameter.
+	WindowLogMin = 10 // from zstd.h
+	// WindowLogMax32 is the maximum value of the windowLog parameter on 32-bit architectures.
+	WindowLogMax32 = 30 // from zstd.h
+	// WindowLogMax64 is the maximum value of the windowLog parameter on 64-bit architectures.
+	WindowLogMax64 = 31 // from zstd.h
+
+	// DefaultWindowLog is the default value of the windowLog parameter.
+	DefaultWindowLog = 0
+)
+
+// A WriterParams allows users to specify compression parameters by calling
+// NewWriterParams.
+//
+// Calling NewWriterParams with a zero-value WriterParams is equivalent to
+// calling NewWriter.
+type WriterParams struct {
+	// Compression level. Special value 0 means default.
+	CompressionLevel int
+
+	// WindowLog. Must be clamped between WindowLogMin and WindowLogMin32/64.
+	// Special value 0 means 'use default windowLog'.
+	//
+	// Note: enabling log distance matching increases memory usage for both
+	// compressor and decompressor. When set to a value greater than 27, the
+	// decompressor requires special treatment.
+	WindowLog int
+
+	// Dict is the dictionnary used for compression. May be nil.
+	Dict *CDict
 }
 
-func newWriterDictLevel(w io.Writer, cd *CDict, compressionLevel int) *Writer {
+// NewWriterParams returns new zstd writer writing compressed data to w
+// using the given set of parameters.
+//
+// The returned writer must be closed with Close call in order
+// to finalize the compressed stream.
+//
+// Call Release when the Writer is no longer needed.
+func NewWriterParams(w io.Writer, params WriterParams) *Writer {
```

Comment:
I'd suggest passing `params` by reference, so it could accept `nil` for default params.

Reply:
done